lib/actions/updateGuest.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  getLastNonProxyCheckin,
+  updateCheckin,
+  updateGuest,
+  getCurrentGuest,
+} from '~lib/db'
+import { queryCache } from 'react-query'
+import * as db from '../db'
+import * as api from '../api'
+import { encrypt, toCSV } from '../crypto'
+
+export async function updateCurrentGuest(guest: db.Guest): Promise<db.Checkin> {
+  // First we update the current guest in local DB (if it exists)
+  const currentGuest = await getCurrentGuest()
+  if (currentGuest) {
+    await updateGuest(guest)
+  }
+  // The we fetch and update the last checkin
+  const currentCheckin = await getLastNonProxyCheckin()
+  currentCheckin.guest = guest
+  const updatedCheckin = await updateCheckin(currentCheckin)
+  // We also need to update the data in the API (only if the public key is available (downwards compability))
+  if (updatedCheckin.k) {
+    const csv = toCSV(updatedCheckin.guest)
+    const encryptedData = encrypt(updatedCheckin.k, csv)
+    await api.patchTicket({
+      id: updatedCheckin.id,
+      encryptedData: encryptedData,
```

Comment:
This won't work because of the `write_once_only` validation https://github.com/railslove/rcvr-api/blob/master/app/models/ticket.rb#L16
This was done to make sure that data is not overwritten. Perhaps we can simply create a new checkin ticket with the new data. What do you think?

Reply:
@salimhb ah damn. I did not see that. Do you think we can checkout the current ticket and then create a new one?

Reply:
Yes, I think that this would be the only way.

Reply:
If you want to update a ticket, then you need to know the UUID. As you said if we create a new record it would look like this: 

<img width="662" alt="Screenshot 2021-02-19 at 22 40 26" src="https://user-images.githubusercontent.com/1360387/108684156-6d4f1280-74ea-11eb-916a-548616af7517.png">

So I would go ahead and change the API. Alternatively we could ask @koos if we really need to update the current ticket in the backend. What you think @salimhb @bonflintstone

Reply:
Your trusting the good will of the guest here ;)
The idea behind making it write-only at first, was to avoid potential security risks. Any audit system requires tracking changes. It's not good to overwrite old data irreversibly.

If we allow rewriting contact data, then it can be maliciously used like this:
1. As a guest, I check in with my correct information.
2. The waiter/owner/police controls that I entered valid data.
3. Before leaving, I edit my contact data with fake info.
4. I checkout.

My ticket now has invalid info, even though it was previously verified.

Reply:
> Alternatively we could ask @koos if we really need to update the current ticket in the backend.
Would that not make this entire PR moot? :smile: 

I am all for changing the backend, I think that's gonna be quite easy. I can also help with that :)